app/controllers/application_api_controller.rb

```diff
@@ -0,0 +1,23 @@
+# frozen_string_literal: true
+
+class ApplicationApiController < ActionController::API
+  include ActionController::HttpAuthentication::Token::ControllerMethods
+
+  private
+
+    # Returns 406 status if format requested is not json. This method should be used as
+    # a before_action callback for any controller actions that only respond to json.
+    def ensure_json_request
+      return if request.format.blank? || request.format == :json
+      head :not_acceptable
+    end
+
+    # Renders with an :unauthorized status if no request token is provided, or renders with a
+    # :forbidden status if the request uses an invalid request token. This method should be
+    # used as a before_action callback for any controller actions that require authorization.
+    def authenticate_request_token
+      authenticate_or_request_with_http_token do |token, _options|
+        DERIVATIVO['remote_request_api_key'] == token
```

Comment:
More of an FYI than a suggested change:
ActiveSupport::SecurityUtils#fixed_length_secure_compare
ActiveSupport::SecurityUtils#secure_compare

I don't have a lot of experience with the above, but they seem to be recommended to prevent timing attacks. Not sure about a possible performance hit.

Reply:
Good suggestion!  Since the Derivativo API isn't hit too hard, and most of the time is spent doing conversions, I don't mind the performance trade-off.  I'll make that change now.  Thanks!